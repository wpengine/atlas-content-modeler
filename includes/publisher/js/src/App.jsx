/* global atlasContentModelerFormEditingExperience */
import React, { useEffect, useState } from "react";
import Fields from "./components/Fields";
import { __ } from "@wordpress/i18n";
import { sendEvent } from "acm-analytics";
import TrashPostModal from "./components/TrashPostModal";

export default function App({ model, mode }) {
	const isEditMode = mode === "edit";
	const [trashPostModalIsOpen, setTrashPostModalIsOpen] = useState(false);

	useEffect(() => {
		const queryParams = new URLSearchParams(window.location.search);
		const newPost = queryParams.get("acm-post-published");
		if (newPost) {
			sendEvent({
				category: "post",
				action: "Post Published",
			});
		}
	}, []);

	return (
		<div className="app classic-form" style={{ marginTop: "20px" }}>
			<div className="flex-parent">
				<div>
					<h3 className="main-title">
						{isEditMode ? (
							<span>{__("Edit", "atlas-content-modeler")} </span>
						) : (
							<span>{__("Add", "atlas-content-modeler")} </span>
						)}
						{model.singular}
					</h3>
				</div>

				{isEditMode && (
					<div
						style={{ marginLeft: "20px" }}
						className="flex-align-v"
					>
						<a
							className="page-title-action"
							href={
								"/wp-admin/post-new.php?post_type=" + model.slug
							}
						>
							Add New
						</a>
					</div>
				)}
			</div>
			<div className="d-flex flex-column">
				<Fields model={model} />
			</div>
			<TrashPostModal
				isOpen={trashPostModalIsOpen}
				setIsOpen={setTrashPostModalIsOpen}
			/>
		</div>
	);
}
