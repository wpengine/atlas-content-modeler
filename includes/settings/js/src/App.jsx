import React from "react";

export default function App() {
	return (
		<p>This is the App component.</p>
	);
}
