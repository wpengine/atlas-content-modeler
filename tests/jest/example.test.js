import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../includes/settings/js/src/App';

describe('Example tests', () => {
	it('is true that true is true', () => {
		expect(true).toBe(true);
	});

	const app = renderer.create(
		<App/>
	);

	let tree = app.toJSON();

	it('React', () => {
		expect(tree).toMatchSnapshot();
	});
});

