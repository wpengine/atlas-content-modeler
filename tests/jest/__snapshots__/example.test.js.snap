// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example tests React 1`] = `
<p>
  This is the App component.
</p>
`;
